src/CollapsableMixin.js

```diff
@@ -1,103 +1,133 @@
 var React = require('react');
 var TransitionEvents = require('./utils/TransitionEvents');
 
+/* This mixin keeps track of its own 'internal' expanded state.  This is to
+ * help enable running css animations on the collapsable component.  The
+ * expanded state /cannot/ bet set /before/ the mixin has had a chance to
+ * ensure that dimensions have been set on the collaspable component.  This
+ * is due to how browers interact with css transitions.  If the mixin does 
+ * not share an 'expanded' state then it makes it a lot easier to handle the
+ * order in which the dimension and `expanded` information is set.
+
+ * This pattern requires the owner of this mixin to manually call the
+ * internalToggle() function to toggle the expanded/collapsed state instead
+ * of relying upon props propigation.
+ *
+ * It is suggested that the component which uses this mixin keep its own
+ * expanded state (which could be from props or state) and call the
+ * internalToggle() function in componentDidUpdate().  Ex:
+
+   componentDidUpdate: function(prevProps, prevState){
+     var wasExpanded = prevState.expanded;
+     var isExpanded = this.state.expanded;
+     if(wasExpanded != isExpanded){
+       this.internalToggle();
+     }
+   }
+*/
```

Comment:
I can't say that I completely understand why we can't continue to allow `expanded` to be used on the components that use this mixin. The point of this mixin is to isolate that logic away from the components that use it. If the mixin needs to retain it's own internal representation of `expanded` then use something internal such as `__collapsableMixinExpanded__`.